### PR TITLE
New post

### PR DESCRIPTION
**Pull request by Tr00d**
Wording update on interview + full review

### CODE REVIEW

File: i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx [hunk 1]
```diff
@@ -45,52 +45,50 @@ _tldr; fais des katas en appliquant le Test Driven Development!_
...
+Évidemment, cela n'aura pas été une adoption immédiate de ma part! Bien que je comprenais pourquoi le testing était important ainsi que tous les bénéfices associés, je pensais que c'était trop dur parce que je ne voyais pas comment introduire des tests dans le projet sur lequel j'étais assigné. Pour être tout à fait honnête: OUI, c'était dur.
+Mais pas parce que c'était du testing: le produit sur lequel je travaillais était en fait un monolithe avec énormément de couplage et le testing aurait intégré un (trop) gros _refactoring_.
 
-Je savais que je devais apprendre des principes et des _patterns_ (principalement sur comment construire une application maintenable) avant d'être capable d'introduire des tests. Ça m'a pris du temps avant que je ne sois vraiment apte à travailler avec une approche conduite par les tests.
+J'ai vite réalisé que je devais progresser sur d'autres sujets avant d'être capable d'introduire des tests. Cela m'a pris du temps avant que je ne sois vraiment apte à travailler avec une approche _test-driven_.
```
Comment (HunteRoi) [inline i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx:67]: réductible*

File: i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx [hunk 2]
```diff
@@ -45,52 +45,50 @@ _tldr; fais des katas en appliquant le Test Driven Development!_
...
 </AnchoredP>
 
-🔹 Sur ton graphe, ça se situe tout à fait à gauche <img style={{ display: "inline" }} src="https://www.pinclipart.com/picdir/big/524-5246191_troll-face-meme-png-troll-face-clipart.png" width="16" height="16" />
-mais la réponse que tu attends, j'imagine que c'est pourquoi. Je ne suis pas entièrement sûr... mais je dirais que l'ordre de gauche à droite n'est pas annodin.
-À mon sens, plus tu avances vers la droite, plus la portée fonctionnelle de ton test est grande.
+🔹 Monsieur est observateur, cela se situe en effet tout à fait à gauche <img style={{ display: "inline" }} src="https://www.pinclipart.com/picdir/big/524-5246191_troll-face-meme-png-troll-face-clipart.png" width="16" height="16" /> mais j'imagine que tu te demanbes pourquoi.
+Je ne suis pas familier avec ce schéma pour être honnête mais je comprends la classification. Il s'agit de la pyramide de testing (_Agile Testing Pyramid_) mais de gauche à droite au lieu de bas en haut. Plus tu seras situé vers la gauche, plus ton scope est petit et plus tes tests seront nombreux et rapides. La logique inverse est appliquée lorsque tu vas dans l'autre sens.
```
Comment (HunteRoi) [inline i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx:86]: supprimable*

File: i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx [hunk 3]
```diff
@@ -45,52 +45,50 @@ _tldr; fais des katas en appliquant le Test Driven Development!_
...
-	ce qu'est test unitaire s'il te plaît?
+	🔸 Ok! Par contre, histoire de ne pas perdre les lecteurs... tu sais
+	expliquer ce qu'est test unitaire s'il te plaît?
 </AnchoredP>
 
-🔹 Oui, bien sûr! Un test unitaire, c'est un test qui vérifie un _unit of code_. Et c'est là que les interprétations divergent.
-Certaines personnes vont voir ça comme un bout de code (ce qui est mon cas). Une méthode, c'est un _unit of code_ et donc quelque chose que l'on peut tester.
-Puis d'autres personnes peuvent voir ça comme un _behavior_. Ce qui n'est pas faux non plus mais ça dépend de comment tu vois la chose.
+🔹 Oui, bien sûr! Un test unitaire, c'est un test qui vérifie un _unit of code_. La notion de _unit of code_ varie en fonction des écoles de testing.
+L'école London (ou Mockist) va voir cela comme le plus petit _chunk of code_, on parlera d'une classe ou d'une méthode. Par contre, l'école Detroit (ou Classicist) va voir cela comme un ensemble de classes ou de méthodes.
```
Comment (HunteRoi) [inline i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx:75]: On ne rajouterait pas des liens vers des explications de ce que sont ces 2 écoles en détails?

File: i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx [hunk 4]
```diff
@@ -127,132 +121,144 @@ Du coup, les tests d'intégration sont importants. Mais je considère, à tort o
...
 </AnchoredP>
 
-🔹 Évidemment, le risque zéro n'existe pas. Ta suite de tests peut avoir un scénario non-couvert.
-Elle peut potentiellement laisser passer des choses parce que tu as soit oublier un test unitaire (un scénario que tu as oublié de prévoir) soit une intégration qui se passe mal entre des composants.
-C'est pas la mort hein!
-
-Admettons: tu prends le cas d'un _refactoring_ qui te génère 10 bugs. Le même _refactoring_ couvert de tests unitaires te génère 1 bug. Il reste 1 bug, c'est pas terrible. Mais, ça ira plus vite de régler 1 bug que 10.
+🔹 Évidemment, le risque zéro n'existe pas. On parle de _test efficiency_. Tu peux très bien avoir des scenarios non-couverts ou des _corner-case_ qui n'ont pas été pensés.
+Mais tout va bien, on ne perd pas le contrôle pour autant! La solution est toute simple: crée un test qui couvre le scope de ton bug (et donc qui échoue), corrige ton bug et vérifie que ton test passe.
+Au final, même si tu auras tout de même des bugs, penses surtout à la quantité de bugs/régressions qui aura été filtrée et corrigée en amont d'un merge.
```
Comment (HunteRoi) [inline i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx:208]: supprimable*

File: i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx [hunk 5]
```diff
@@ -299,124 +305,139 @@ public class CalculatorTests
...
+Pour couronner le tout, il y a cette courbe de progression dont on a parlé. Tu ne peux pas dire à ton équipe du jour au lendemain : "allez hop, à partir d'aujourd'hui, TDD pour tout le monde!". Ce n'est pas réaliste.
 
-Tu es déjà sur un flux tendu parce que tu n'as pas fait tes tests en amont, ce qui fait que tu te tapes toutes les conséquences dans la figure.
-Et donc, inciter une adoption aux tests unitaires et à TDD devient d'autant plus compliqué parce que les gens sont déjà totalement noyés.
+Donc, tenter une adoption aux tests et à TDD tard dans un développement devient d'autant plus compliqué parce que les gens sont déjà totalement étouffés par le travail et la pression. Le testing sera malheureusement placé tout en bas de la liste des priorités.
+
+Je pense que je suis en train de faire fuir tes lecteurs ou de les démoraliser. J'ai l'air extrêmement pessimiste mais tout n'est pas perdu. J'ai simplement croisé plus de projets sans tests qu'avec. Le changement de dynamique doit venir de nous, développeurs, et cela passe par une prise de conscience de la majorité.
```
Comment (HunteRoi) [inline i18n/fr/docusaurus-plugin-content-blog/2022-01-01-unit-testing.mdx:347]: supprimable*

**Review by HunteRoi — APPROVED** [review]
Hello! Merci pour le travail effectué!!

J'ai ajouté des commentaires inline (soit `supprimable*` soit `réductible*`) pour ce que je pense qu'on peut raccourcir ou supprimer sans perdre beaucoup de contenu ou de sens dans l'article.

Évidemment, comme je l'ai déjà dit: si les gens veulent se poser et lire une partie puis reprendre la lecture plus tard, rien ne les empêche de le faire donc je ne pense pas qu'on devrait couper dans l'article. 

Tout ça pour dire: à toi de voir si on élague ou si on garde comme ça!

a+

**Comment by Tr00d** [comment]
@HunteRoi J'ai appliqué les changements. Je te laisse corriger au besoin.

**Review by HunteRoi — APPROVED** [review]
